src/OOVPADatabase/OOVPA.h

```diff
@@ -200,118 +231,234 @@ typedef enum _eDBScanType {
+#define REGISTER_OOVPA_16(Symbol, Version, ...) REGISTER_OOVPA(Symbol, Version), MSVC_EXPAND(REGISTER_OOVPA_15(Symbol, __VA_ARGS__))
+#define REGISTER_OOVPA_17(Symbol, Version, ...) REGISTER_OOVPA(Symbol, Version), MSVC_EXPAND(REGISTER_OOVPA_16(Symbol, __VA_ARGS__))
+#define REGISTER_OOVPA_18(Symbol, Version, ...) REGISTER_OOVPA(Symbol, Version), MSVC_EXPAND(REGISTER_OOVPA_17(Symbol, __VA_ARGS__))
+#define REGISTER_OOVPA_19(Symbol, Version, ...) REGISTER_OOVPA(Symbol, Version), MSVC_EXPAND(REGISTER_OOVPA_18(Symbol, __VA_ARGS__))
 
-// clang-format off
 #define OV_BYTES_0(...)
- #define OV_BYTES_1(Offset, Value) OV_BYTE(Offset, Value)
- #define OV_BYTES_2(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_1(Offset+1, __VA_ARGS__))
- #define OV_BYTES_3(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_2(Offset+1, __VA_ARGS__))
- #define OV_BYTES_4(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_3(Offset+1, __VA_ARGS__))
- #define OV_BYTES_5(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_4(Offset+1, __VA_ARGS__))
- #define OV_BYTES_6(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_5(Offset+1, __VA_ARGS__))
- #define OV_BYTES_7(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_6(Offset+1, __VA_ARGS__))
- #define OV_BYTES_8(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_7(Offset+1, __VA_ARGS__))
-// clang-format on
+#define OV_BYTES_1(Offset, Value)      OV_BYTE(Offset, Value)
+#define OV_BYTES_2(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_1(Offset + 1, __VA_ARGS__))
+#define OV_BYTES_3(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_2(Offset + 1, __VA_ARGS__))
+#define OV_BYTES_4(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_3(Offset + 1, __VA_ARGS__))
+#define OV_BYTES_5(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_4(Offset + 1, __VA_ARGS__))
+#define OV_BYTES_6(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_5(Offset + 1, __VA_ARGS__))
+#define OV_BYTES_7(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_6(Offset + 1, __VA_ARGS__))
+#define OV_BYTES_8(Offset, Value, ...) OV_BYTE(Offset, Value), MSVC_EXPAND(OV_BYTES_7(Offset + 1, __VA_ARGS__))
+
+#define CALL_unk   call_unknown
+#define CALL_cde   call_cdecl
+#define CALL_std   call_stdcall
+#define CALL_thi   call_thiscall
+#define CALL_fas   call_fastcall
+#define CALL(Name) CALL_##Name
+
+// For generate symbol's suffix name, mainly for registers, and extend API usage.
+#define PARAM(Param, Name)            Param, Name
+#define PARAM1(Param)                 Param, ""
+#define PARAM_TOKEN_unk(Index, Name)  _unk##Index
+#define PARAM_TOKEN_void(Index, Name) // No argument, do not append to symbol reference.
+#define PARAM_TOKEN_stk2(Index, Name) // Argument is stored in call stack, do not append to symbol reference.
```

Comment:
Why is there a second stack argument macro if it is identical to the first one below?

Reply:
Some parameters like, for example, `LARGE_INTEGER` structure is a size of 8 bytes. So, it pushed double stack, aka 8 bytes (or 64 bits). This is necessary if planning to use header files with `LARGE_INTEGER` structure.

And below should be a dead giveaway of their difference.
https://github.com/Cxbx-Reloaded/XbSymbolDatabase/blob/a167837da6773ca1eeb06b7cef6c88d281d15443/include/libParamTypes.h#L5-L6

If the above doesn't answer to your question, what are you not sure, or thinking, about?

Reply:
I understand now, but I believe the name is too vague then. I suggest to rename them to stk32 and stk64 to better clarify their usage/meaning.

Reply:
Then that would require replace whole lot of stk and few stk2 inside `PARAM()` argument. 🙁In my opinion, it will also lead some confusion. Like why is there 32 worth of stack pushed onto call stacks when there's only one stack pushed onto call stack? The same for 64 suffix. If we do it this way, then all of the registers will need to follow this way.

Forked ghidra-xbe currently check for stk#, if there is a number at the end. It will be multiply by 4.

Reply:
What about stk_l and stk_h then, which would stand for low and high respectively (as in low and high 32 bits)? Alternatively, we could also use stk and stk_h to avoid having to replace the many existing stk occurrences in the code.

Reply:
What if we found out there are 3, 4, or more pushes onto stack? What would we do about this? For me, it will be stk# to keep things simple and easier to update.

Reply:
As a final suggestion, you could at least update the comments to say the number of the push, as in `stk -> // ... , 1st push` and `stk2 -> // ... , 2nd push`, and so on if any more are encountered in the future.

Reply:
Per discord discussion further, I realized my error of using `stk` terminology when should have used push, or `psh(#)` by team votes. I made a new commit to replace stk(#) with psh(#) since the attempt to fix past commits isn't suitable.